GVRf/Framework/framework/src/main/java/org/gearvrf/io/GVRMouseDeviceManager.java

```diff
@@ -75,9 +80,17 @@ void removeCursorController(GVRBaseController controller) {
             thread.quitSafely();
             thread = new EventHandlerThread(THREAD_NAME);
             threadStarted = false;
+            if(gvrContext != null){
+                gvrContext.unregisterDrawFrameListener(this);
+            }
         }
     }
 
+    @Override
+    public void onDrawFrame(float frameTime) {
+        thread.invalidate();
```

Comment:
While it would be easy enough to add aforementioned callbacks and also tempting, we need to revisit the whole listeners and events situation since we cannot keep adding stuff.

For now what you could do is:
1) use GVRNotifications on a background thread
2) use onDrawFrameListener as a notification and let the thread pool execute the job